mongo/tests/test_mongo.py

```diff
@@ -185,6 +186,33 @@ def test_mongo_custom_query_with_empty_result_set(aggregator, check, instance_us
     aggregator.assert_metric('dd.custom.mongo.query_a.amount', count=0)
 
 
+def test_mongo_replset(instance_shard, aggregator, check):
+    mongo_check = check(instance_shard)
+    mongo_check.check(None)
+
+    replset_metrics = [
+        'mongodb.replset.health',
+        'mongodb.replset.replicationlag',
+        'mongodb.replset.state',
+        'mongodb.replset.votefraction',
+        'mongodb.replset.votes',
+    ]
+    replset_common_tags = [
+        "replset_name:shard01",
+        "server:mongodb://localhost:27018/",
+        "sharding_cluster_role:shardsvr",
+    ]
+    for metric in replset_metrics:
+        aggregator.assert_metric(metric, tags=replset_common_tags + ['replset_state:primary'])
+    aggregator.assert_metric(
+        'mongodb.replset.optime_lag', tags=replset_common_tags + ['replset_state:primary', 'member:shard01a:27018']
+    )
+    aggregator.assert_metric(
+        'mongodb.replset.optime_lag', tags=replset_common_tags + ['replset_state:secondary', 'member:shard01b:27018']
+    )
+    aggregator.assert_metrics_using_metadata(get_metadata_metrics(), check_submission_type=True)
```

Comment:
Can we also add this to `test_mongo` and `test_mongo2` to see if we missed any other metric?

Reply:
Will be added in #8205